packager/VisibilityChecker.cc

```diff
@@ -477,13 +514,11 @@ class VisibilityCheckerPass final {
                         "but must still respect its enclosing package's imports.",
                         "# packaged: false");
                 }
-
-                if (!db.errorHint().empty()) {
-                    e.addErrorNote("{}", db.errorHint());
-                }
             }
         } else if (*importType == core::packages::ImportType::Test && !this->insideTestFile) {
             // We used a symbol from a `test_import` in a non-test context
+            // TODO(neil): we need to do the above "can't import if layering violation/strict_deps violation" check here
+            // too
```

Comment:
Doing this will require a bunch of refactoring, so going to do this in a separate PR to prevent this one from getting from more messy.